planner/core/src/test/java/eu/seaclouds/platform/planner/core/DamGeneratorTest.java

```diff
@@ -124,5 +125,112 @@ public void testGroupsAsTopologyChild() throws Exception {
         assertTrue(topologyGroups.containsKey("sla_gen_info"));
     }
 
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testNuroDAM() throws Exception {
+        String adp = new Scanner(new File(Resources.getResource("nuro_adp.yml").toURI())).useDelimiter("\\Z").next();
+
+        when(fakeAgreementManager.generateAgreeemntId(((Map<String, Object>) anyObject())))
+                .thenReturn(FAKE_AGREEMENT_ID);
+
+        DamGenerator damGenerator = new DamGenerator(MONITOR_URL, MONITOR_PORT, SLA_ENDPOINT, INFLUXDB_URL, INFLUXDB_PORT);
+        damGenerator.setAgreementManager(fakeAgreementManager);
+        dam = damGenerator.generateDam(adp);
+        template = (Map<String, Object>) yamlParser.load(dam);
+
+        String expectedDAMString = new Scanner(new File(Resources.getResource("nuro_dam.yml").toURI())).useDelimiter("\\Z").next();
+        Map<String, Object> expectedDAM = (Map<String, Object>) yamlParser.load(expectedDAMString);
```

Comment:
Why we have two different objects with the same information, `template` and `expectedDAM`?
Please, use `Dam`, take a look [here](http://stackoverflow.com/questions/15526107/acronyms-in-camelcase)